libraries/ESP8266WebServer/src/ESP8266WebServer.h

```diff
@@ -68,15 +69,17 @@ class FS;
 
 class ESP8266WebServer
 {
+  friend class ESP8266WebServerSecure; // HTTPS server needs access to privates
```

Comment:
Wouldn't it be better to make the privates protected instead of using friend class?

Reply:
@stackfull At the time of writing, I wanted to minimize any impact it would have on the parent class like changing the semantics of existing member variables/functions.

Practically, though, you've got a point here.  I'll have to do some testing first but I'll try and update this PR later in the week.